### PR TITLE
Site Editor: refactor close button slot

### PR DESCRIPTION
**Pull request by vindl**
## Description

This is a follow up to https://github.com/WordPress/gutenberg/pull/22001 that addresses some of the outstanding feedback there. A couple of things were changed when compared to the previous version:

- The soon to be deprecated regular slot was replaced with the one that uses `bubblesVirtually` in order to make this compliant with https://github.com/WordPress/gutenberg/pull/22027
- Changed the custom `wordPressLogo` icon with the one from the `@wordpress/icons` package. This was done in order to align it with latest changes in post editor.
- Added docs for `MainDashboardButton`
- Provides a way to customize only the icon while retaining the rest of the functionality

For the last item I experimented with a couple of things. I didn't see a way to make it work with fill props since we are replacing the whole component that we want to customize if the fill is present. I tried adding a separate slot just for the icon. While that works it seems unnecessarily convoluted. In the end I settled for exporting the existing close button component from the `edit-site` package so it can be customized via props and used as a fill.

## How has this been tested?

If you want to replace the whole component:

```javascript
import { registerPlugin } from '@wordpress/plugins';
import { __experimentalMainDashboardButton as MainDashboardButton } from '@wordpress/edit-site'

registerPlugin( 'main-dashboard-button-test', {
    render() {
        return (
            <MainDashboardButton>
	        <CustomCloseButton />
	    </MainDashboardButton>
        );
    },
} );
```

If you just want to customize the icon:

```javascript
import { registerPlugin } from '@wordpress/plugins';
import { 
__experimentalMainDashboardButton as MainDashboardButton,
__experimentalFullscrenModeClose as FullscrenModeClose } from '@wordpress/edit-site'
import { close } from '@wordpress/icons';

registerPlugin( 'main-dashboard-button-test', {
    render() {
        return (
            <MainDashboardButton>
	        <FullscreenModeClose icon={ close } />
	    </MainDashboardButton>
        );
    },
} );
```

## Screenshots 

| Before | After |
| - | - |
|Screenshot 2020-05-07 at 18 07 12|Screenshot 2020-05-07 at 18 06 00|

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [ ] I've updated all React Native files affected by any refactorings/renamings in this PR.

### CODE REVIEW

File: packages/edit-site/src/components/header/fullscreen-mode-close/index.js
Comment (epiqueras) [inline packages/edit-site/src/components/header/fullscreen-mode-close/index.js:23]:
```suggestion
	const buttonIcon = icon || wordpress;
```
When would an icon ever be falsy?

File: packages/edit-site/src/components/header/index.js
Comment (epiqueras) [inline packages/edit-site/src/components/header/index.js:38]: This should be defined in the slot's file.

**Review by epiqueras — CHANGES_REQUESTED** [review]

File: packages/edit-site/src/components/header/fullscreen-mode-close/index.js
Comment (vindl) [reply packages/edit-site/src/components/header/fullscreen-mode-close/index.js]: Updated in cbc8eba28bdb0ae229cbc7a54c0e4190eff813e9.

File: packages/edit-site/src/components/header/index.js
Comment (vindl) [reply packages/edit-site/src/components/header/index.js]: Moved in cbc8eba28bdb0ae229cbc7a54c0e4190eff813e9.

**Review by epiqueras — APPROVED** [review]

File: docs/designers-developers/developers/slotfills/main-dashboard-button.md
Comment (youknowriad) [inline docs/designers-developers/developers/slotfills/main-dashboard-button.md:10]: so this slot is just an edit-site slot? Would we need to offer the same for the other screens. Also, if the fallback button is the same across screens, should the slot be the same across screens too?

Comment (youknowriad) [reply docs/designers-developers/developers/slotfills/main-dashboard-button.md:10]: This shouldn't block this PR. I think it's ready.

Comment (vindl) [reply docs/designers-developers/developers/slotfills/main-dashboard-button.md:10]: > so this slot is just an edit-site slot?

Yes.

> Would we need to offer the same for the other screens. Also, if the fallback button is the same across screens, should the slot be the same across screens too?

I was planning to make a separate slot for post editor. The rationale being that someone might want to provide a different component in other screens. Or it might look the same but we might want it to behave differently.

Comment (youknowriad) [reply docs/designers-developers/developers/slotfills/main-dashboard-button.md:10]: I'm not 100% certain we need a separate component especially if we add support to more and more screens (widgets...). Also it seems possible to enqueue the extension based on the screen or check the screen and change the behavior.

Comment (vindl) [reply docs/designers-developers/developers/slotfills/main-dashboard-button.md:10]: > Also it seems possible to enqueue the extension based on the screen or check the screen and change the behavior.

That's true. Since I was planning on starting a separate PR for post editor, would it be fine if I consolidate this to use the same slot in it?

Comment (youknowriad) [reply docs/designers-developers/developers/slotfills/main-dashboard-button.md:10]: Yes, we should merge this and see in a separate PR if we can have the same slot.

**Pull request merged**
